wgsl/index.bs

```diff
@@ -942,9 +1023,281 @@ a result value, such as the non-result-value effects of its subexpressions.
+  <thead>
+    <tr><th>Src
+        <th>Dest
+        <th>ConversionRank(Src,Dest)
+        <th>Notes
+  </thead>
+  <tr algorithm="conversion rank identity">
+      <td>|T|
+      <td>|T|
+      <td>0
+      <td>Identity. No conversion performed.
+  <tr algorithm="conversion rank from reference via load rule">
+      <td>ref&lt;|S|,|T|,|A|&gt;<br>where |A| is [=access/read=] or [=access/read_write=]
+      <td>|T|
+      <td>0
+      <td>Apply the [=Load Rule=] to load a value from a memory reference.
+  <tr algorithm="conversion rank abstract float to f32">
+      <td>[=AbstractFloat=]
+      <td>f32
+      <td>1
+      <td>
+  <tr algorithm="conversion rank abstract int to i32">
+      <td>[=AbstractInt=]
+      <td>i32
+      <td>2
+      <td>
+  <tr algorithm="conversion rank abstract int to u32">
+      <td>[=AbstractInt=]
+      <td>u32
+      <td>3
+      <td>
+  <tr algorithm="conversion rank abstract int to abstract float">
+      <td>[=AbstractInt=]
+      <td>[=AbstractFloat=]
+      <td>4
+      <td>
+  <tr algorithm="conversion rank abstract int to f32">
+      <td>[=AbstractInt=]
+      <td>f32
+      <td>5
+      <td>Behaves as [=AbstractInt=] to [=AbstractFloat=], and then [=AbstractFloat=] to f32
+  <tr algorithm="conversion rank for non-convertible cases">
+      <td>|S|
+      <td>|T|<br>where above cases don't apply
+      <td>infinity
+      <td>There are no automatic conversions between other types.
+</table>
+
+### Overload Resolution ### {#overload-resolution-section}
+
+When more than one [=type rule applies to a syntactic phrase=], a tie-breaking procedure is used
+to determine which one should take effect.
+This procedure is called <dfn noexport>overload resolution</dfn>,
+and assumes type checking has already succeeded in finding static types for [=subexpressions=].
+
+Consider a [=syntactic phrase=] |P|, and all [=type rule applies to a syntactic phrase|type rules that apply=] to |P|.
+The overload resolution algorithm calls these type rules <dfn noexport>overload candidates</dfn>.
+For each candidates:
+* Its preconditions have been met.
+* Its [=type rule conclusion|conclusion=] has:
+    * A syntactic form matching a valid parse of |P|, and
+    * A [=type assertion=] corresponding to each [=top-level expression=] in |P|.
+
+Overload resolution for |P| proceeds as follows, with the goal of finding a single most preferable [=overload candidate=]:
+
+1. For each candidate |C|, enumerate conversion ranks for subexpressions in the syntactic phrase.
+    The candidate's preconditions have been met, and so for the |i|'th subexpression in the |P|:
+    * Its static type has computed.
+    * There is a [=feasible automatic conversion=] from the expression's static type to the type required by the corresponding type assertion in the preconditions.
+        Let |C|.|R|(i) be the [=ConversionRank=] of that conversion.
+
+1. Rank candidates: Given two overload candidates |C1| and |C2|, |C1| is <dfn>better</dfn> than |C2| if:
```

Comment:
"Better" seems like an odd thing to define for such a specific case. Maybe "preferable candidate" instead?